File: client/src/app/models/reservation.ts
export interface Reservation {
  id: number;
  reservableObjectId: number;
  objectName: string;
  objectType: string;
  startDateTime: string;
  endDateTime: string;
  createdAt: string;
}

export interface CreateReservationRequest {
  reservableObjectId: number;
  startDateTime: string;
  endDateTime: string;
}

export interface AvailabilityResponse {
  available: boolean;
}

File: api/src/Workshop.Api/Data/Entities/Reservation.cs
namespace Workshop.Api.Data.Entities;

public class Reservation
{
    public int Id { get; set; }
    public int ReservableObjectId { get; set; }
    public string UserId { get; set; } = string.Empty;
    public DateTime StartDateTime { get; set; }
    public DateTime EndDateTime { get; set; }
    public DateTime CreatedAt { get; set; }

    public ReservableObject? ReservableObject { get; set; }
}
